apps/client/src/context/AuthContext.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from 'react';
+import {
+  SignInWithPasswordCredentials,
+  SignUpWithPasswordCredentials,
+  User,
+} from '@supabase/supabase-js';
+import { supabase } from '../supabase/supabseClient';
+import { isNotNullOrUndefined } from '../utils/arrays';
+
+type AuthContextType = {
+  signUp: typeof supabase.auth.signUp;
+  signIn: typeof supabase.auth.signInWithPassword;
+  signOut: typeof supabase.auth.signOut;
+  user: User | null;
+};
+
+// TODO: figure out how to use this type without adding "isUserChangedEventType" function
+// type UserChangedEventType = Extract<AuthChangeEvent, 'SIGNED_IN' | 'USER_UPDATED' | 'TOKEN_REFRESHED'>;
+const UserChangedEvents = ['SIGNED_IN', 'USER_UPDATED', 'TOKEN_REFRESHED'];
+const SIGN_OUT_EVENT = 'SIGNED_OUT';
+
+export const AuthContext = React.createContext<AuthContextType | null>(null);
+
+export const AuthProvider: React.FC<React.PropsWithChildren> = ({ children }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const loadUserFromLocalStorage = async () => {
+      const {
+        data: { session },
+      } = await supabase.auth.getSession();
+
+      if (isNotNullOrUndefined(session)) {
+        setUser(session.user ?? null);
+      }
+
+      setIsLoading(false);
+    };
+
+    loadUserFromLocalStorage();
+  }, []);
```

Comment:
Damn didn't think about this
But since I'm accessing the local storage and not making API call, can I still use it?
I wanted the `isLoading`  just to not show anything while accessing the localStorage (In theory to "block" users from seeing unauthorized stuff) but I guess I can delete it because its pretty damn fast